### PR TITLE
[OF-1313] feat: Add name property to component base

### PR DESCRIPTION
**Pull request by MAG-mv**
- Added GetComponentName and SetComponentName to ComponentBase
- Deprecated Name properties and accessors on derived classes
- Setup script accessor for aname
- Added tests for name in ComponentBase

### CODE REVIEW

File: Library/src/Multiplayer/Script/ComponentScriptInterface.cpp [hunk 1]
```diff
@@ -64,6 +64,24 @@ int64_t ComponentScriptInterface::GetComponentType() const
 	return (int64_t) ComponentType::Invalid;
 }
 
+void ComponentScriptInterface::SetComponentName(std::string name)
+{
+	if (Component)
```
Comment (MAG-SamBirley) [inline Library/src/Multiplayer/Script/ComponentScriptInterface.cpp:69]: Should we emit an error log if component == nullptr for the setter and getter?

File: Library/src/Multiplayer/ComponentBase.cpp [hunk 2]
```diff
@@ -18,6 +18,7 @@
 #include "CSP/Common/List.h"
 #include "CSP/Multiplayer/Script/EntityScript.h"
 #include "CSP/Multiplayer/SpaceEntity.h"
+#include "ComponentBaseKeys.h"
```
Comment (MAG-SamBirley) [inline Library/src/Multiplayer/ComponentBase.cpp:21]: It looks like this header is not part of this PR. I can see it's causing failures [on the PR checks](https://uk-teamcity.magnopus.global/buildConfiguration/Olympus_Foundation_X64_ReleaseDLLCSharp/229426?buildTab=log&linesState=312.13556&logView=flowAware&focusLine=14061).

**Review by MAG-SamBirley — CHANGES_REQUESTED** [review]

File: Library/src/Multiplayer/Script/ComponentScriptInterface.cpp (hunk 1 above)
Comment (MAG-mv) [reply Library/src/Multiplayer/Script/ComponentScriptInterface.cpp:69]: We currently dont do this for any other script component props

Comment (MAG-SamBirley) [reply Library/src/Multiplayer/Script/ComponentScriptInterface.cpp:69]: Understood. Do you think we should?

Comment (MAG-mv) [reply Library/src/Multiplayer/Script/ComponentScriptInterface.cpp:69]: We should further investigate this by understanding the circumstances where the COmponent would be null to find the best method to handle this

Comment (MAG-SamBirley) [reply Library/src/Multiplayer/Script/ComponentScriptInterface.cpp:69]: Makes sense. Thanks!

File: Tests/src/PublicAPITests/ComponentTests/ComponentTests.cpp [hunk 3]
```diff
@@ -14,6 +14,9 @@
  * limitations under the License.
  */
 
+#include "../SpaceSystemTestHelpers.h"
```
Comment (MAG-IY) [inline Tests/src/PublicAPITests/ComponentTests/ComponentTests.cpp:17]: Should these paths not be relative to the project root like the rest? Or is there a reason for the file relative path?

Comment (MAG-mv) [reply Tests/src/PublicAPITests/ComponentTests/ComponentTests.cpp:17]: image
Component tests is in a nested folder and we dont have an include dir setup at the root

Comment (MAG-IY) [reply Tests/src/PublicAPITests/ComponentTests/ComponentTests.cpp:17]: OK fair enough. Thought that might be your answer, but just wanted to check. It's not something I feel strongly about anyway.

**Review by MAG-SamBirley — APPROVED** [review]
Looks good to me.

One minor suggestion for the future would be to note in the new base keys header that adding extra default values requires updating the component tests (where the number of default values is hard coded). Not significant enough to block the PR though.

**Pull request merged**